Missing Semigroup instance

Missing Semigroup instances prevents the library to be compile on recent ghc versions:
```
Database/Beam/Postgres/PgSpecific.hs:433:39: error:
        • Could not deduce (Semigroup (PgJSONB a))
            arising from the 'deriving' clause of a data type declaration
          from the context: Monoid a
            bound by the deriving clause for ‘Monoid (PgJSONB a)’
            at Database/Beam/Postgres/PgSpecific.hs:433:39-44
          Possible fix:
            use a standalone 'deriving instance' declaration,
              so you can specify the instance context yourself
        • When deriving the instance for (Monoid (PgJSONB a))
        |
    433 |   deriving ( Show, Eq, Ord, Hashable, Monoid )
        |                                       ^^^^^^
```

--- a/beam-postgres/Database/Beam/Postgres/PgSpecific.hs
+++ b/beam-postgres/Database/Beam/Postgres/PgSpecific.hs
@@ -405,7 +405,7 @@ arrayOf_ q =
 -- The 'defaultMigratableDbSettings' function automatically assigns the postgres
 -- @JSON@ type to fields with this type.
 newtype PgJSON a = PgJSON a
-  deriving ( Show, Eq, Ord, Hashable, Monoid )
+  deriving ( Show, Eq, Ord, Hashable, Monoid, Semigroup )
 
 instance HasSqlEqualityCheck PgExpressionSyntax (PgJSON a)
 instance HasSqlQuantifiedEqualityCheck PgExpressionSyntax (PgJSON a)
@@ -430,7 +430,7 @@ instance ToJSON a => HasSqlValueSyntax PgValueSyntax (PgJSON a) where
 --
 -- Fields with this type are automatically given the Postgres @JSONB@ type
 newtype PgJSONB a = PgJSONB a
-  deriving ( Show, Eq, Ord, Hashable, Monoid )
+  deriving ( Show, Eq, Ord, Hashable, Monoid, Semigroup )
 
 instance HasSqlEqualityCheck PgExpressionSyntax (PgJSONB a)
 instance HasSqlQuantifiedEqualityCheck PgExpressionSyntax (PgJSONB a)

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,6 +1,6 @@
 #resolver: lts-9.0
 #resolver: lts-10.3
-resolver: lts-11.2
+resolver: lts-11.10
 packages:
   - beam-sqlite
   - beam-postgres

